Microservices.PlatformService/Controllers/PlatformController.cs

```diff
@@ -0,0 +1,151 @@
+
+            var platformReadDto = await _service.CreateAsync(platform);
+
+            // Send Sync Message
+            try
+            {
+                await _commandDataClient.SendPlatformToCommand(platformReadDto);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"--> Could not send by HttpClient: {ex.Message}");
+            }
+
+            // Send async message to Message Bus
+            try
+            {
+                var platformPublishDto = _mapper.Map<PlatformPublishDto>(platformReadDto);
+                platformPublishDto.Event = "Platform_Published";
+                _messageBusClient.PublishPlatform(platformPublishDto);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"--> Could not send by RabbitMq: {e.Message}");
+            }
+
+            return CreatedAtRoute(nameof(GetPlatformById), new { platformReadDto.Id }, platformReadDto);
+        }
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<PlatformReadDto>> UpdatePlatform(PlatformReadDto platform)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            PlatformReadDto platformReadDto;
+            try
+            {
+                platformReadDto = await _service.UpdateAsync(platform);
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+
+            return CreatedAtRoute(nameof(GetPlatformById), new { platformReadDto.Id }, platformReadDto);
+        }
+
+        [HttpDelete("{id:int}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<ActionResult<PlatformReadDto>> DeletePlatform(int id)
+        {
+            PlatformReadDto platformReadDto;
+            try
+            {
+                platformReadDto = await _service.GetByIdAsync(id);
+            }
+            catch (Exception e)
+            {
+                return e switch
+                {
+                    ArgumentNullException => NotFound(
+                        $"Exception type - {nameof(ArgumentNullException)} \nException message - {e.Message}"),
+                    ArgumentException => NotFound(
+                        $"Exception type - {nameof(ArgumentException)} \nException message - {e.Message}"),
+                    _ => BadRequest($"Exception type - {nameof(Exception)} \nException message - {e.Message}")
+                };
+            }
+
+            await _service.DeleteAsync(id);
+
+            return CreatedAtRoute(nameof(GetPlatformById), new { Id = id }, platformReadDto);
```

Comment:
Why does DeletePlatform action return CreatedAtRoute result?

Reply:
Sure. missed this part. But I wan't to return modal from delete endpoint.